Dip/Dip/Definition.swift

```diff
@@ -50,20 +50,62 @@ func ==(lhs: DefinitionKey, rhs: DefinitionKey) -> Bool {
 public enum ComponentScope {
   /// Indicates that a new instance of the component will be created each time it's resolved.
   case Prototype
+  /// Indicates that instances will be reused during resolve but will be discurded when topmost `resolve` method returns.
+  case ObjectGraph
   /// Indicates that resolved component should be retained by container and always reused.
   case Singleton
 }
 
 ///Definition of type T describes how instances of this type should be created when they are resolved by container.
 public final class DefinitionOf<T>: Definition {
```

Comment:
Haven't grasped that one yet. It seems like it has the same issue (mutating postfix function), but not sure how to avoid it in this case or if that's as problematic.
I guess I'm less annoyed by this case somehow. Maybe we could add a security in that `resolveDependencies` method so that it `fatalError` if we try to call it while it already has a value and so has already been called before, to prevent multiple calls at least?